src/htm/regions/InputRegion.cpp

```diff
@@ -0,0 +1,65 @@
+/* ---------------------------------------------------------------------
+ * HTM Community Edition of NuPIC
+ * Copyright (C) 2019, Numenta, Inc.
+ *
+ * Author: David Keeney, July 2020, dkeeney@gmail.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ * --------------------------------------------------------------------- */
+
+/** @file
+ * Implementation of a network region that handles direct input to a region.
+ * This is not a normal region.  It does not need to be declared with an addRegion
```

Comment:
this is a good explanation :+1: 
move it to the hpp?

Reply:
I am not very consistent about where I put text like this.  Sure, I will move it.